### PR TITLE
Add ingest service requests forwarding

### PR DESCRIPTION
**Pull request by guilload**
### Description
Forward ingest requests from non-indexer nodes to indexers.

Todo:
- [x] fix `IngestServiceError` conversion to `tonic::Status`
- [x] fix `AskError<_>` conversion to IngestServiceError

### How was this PR tested?
- [x] added unit test
- [x] tested locally with 2 nodes

### CODE REVIEW

File: quickwit/quickwit-codegen/src/codegen.rs [hunk 1]
```diff
@@ -311,21 +375,32 @@ fn generate_tower_mailbox(mailbox_name: &Ident, error_path: &syn::Path) -> Token
...
         {
             type Response = T;
-            type Error = #error_path;
+            type Error = #error_type;
             type Future = BoxFuture<Self::Response, Self::Error>;
 
             fn poll_ready(&mut self, _cx: &mut std::task::Context<'_>) -> std::task::Poll<Result<(), Self::Error>> {
-                //! This does not work with balance middlewares such as `tower::balance::pool::Pool` because this always returns `Poll::Ready`.
-                //! The fix is to acquire a permit from the mailbox in `poll_ready` and consume it in `call`.
+                //! This no longer works with parallel actors. The fix is to acquire a permit from the mailbox in `poll_ready` and consumer it in `call`.
```
Comment (fmassot) [inline quickwit/quickwit-codegen/src/codegen.rs:403]: why did you change the comment? I like the previous one.

File: quickwit/quickwit-serve/src/ingest_api/rest_handler.rs [hunk 2]
```diff
@@ -91,12 +86,14 @@ struct BulkActionMeta {
     id: String,
 }
 
+fn bind<T: Clone + Send>(value: T) -> impl Filter<Extract = (T,), Error = Infallible> + Clone {
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/ingest_api/rest_handler.rs:89]: we can probably make it `pub(crate)` and put it elsewhere (can handle this in another PR)

File: quickwit/quickwit-codegen/src/codegen.rs (hunk 1 above)
Comment (guilload) [reply quickwit/quickwit-codegen/src/codegen.rs:403]: Rebase gone wrong. I'll fix :)

**Review by fmassot — APPROVED** [review]
GG
